src/frontend/src/btc/schedulers/btc-wallet.scheduler.ts

```diff
@@ -45,16 +65,30 @@ export class BtcWalletScheduler implements Scheduler<PostMessageDataRequestBtc>
 			network: bitcoinNetwork
 		});
 
-		// TODO: Fetch and parse transactions
-		const transactions: BitcoinTransaction[] = [];
+		let uncertifiedTransactions: CertifiedData<BitcoinTransaction>[] = [];
+
+		if (data?.shouldFetchTransactions) {
+			const btcAddress =
+				data?.btcAddress ??
```

Comment:
What happens if the the new requested address is different than the address in the worker?
Is there any comparison of the certification, what happens if one address is certified and the other no? Or are addresses always certified? Do we have an assertion to ensure the address is certified?

Reply:
`this.btcAddress` is always certified. After double-checking, it appeared that `data?.btcAddress` can indeed be uncertified if it's loaded from IDB and not canister directly.  

I updated the code to always fetch certified BTC address on the first worker run. After I'm done with the main functionality, I'll come back to the approach of trying to get the address from FE (while keeping in mind the questions you raised). For that, I added an additional TODO comment.

Reply:
> I updated the code to always fetch certified BTC address on the first worker run. 

This will also be the call that will certify that the address in IDB is certified or that will be an extra call?

Context: we need update call for security to ensure data are valid but, we also need to spare update call because it loads the network. 🤪

Reply:
1. There is no guarantee that FE loads the address before BTC wallet worker is launched. At least locally, I had an impression that it never happens. 
2. FE loads the address from IDB, and only after that certifies it. 

So, in order not to wait for those steps done by FE, I'm loading BTC address in the worker. Later, we can improve it by syncing these 2 processes (whoever loads it first, notifies the other one), similar as it's done in the IC wallet worker,

Reply:
Thanks!